### PR TITLE
use snippet search optionally

### PR DESCRIPTION
**Pull request by sam-h-bean**
Add an additional flag which allows for hitting our new endpoint.

### CODE REVIEW

File: libs/langchain/langchain/retrievers/you.py [hunk 1]
```diff
@@ -17,6 +17,7 @@ class YouRetriever(BaseRetriever):
     """
 
     ydc_api_key: str
+    endpoint_type: Optional[str] = "web"
```
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/you.py:20]: ooc why make this Optional?

File: libs/langchain/langchain/retrievers/you.py [hunk 2]
```diff
@@ -34,13 +35,20 @@ def _get_relevant_documents(
...
+                f"https://api.ydc-index.io/search?query={query}",
+                headers=headers,
+            ).json()
+
+            docs = []
+            for hit in results["hits"]:
+                for snippet in hit["snippets"]:
+                    docs.append(Document(page_content=snippet))
+            return docs
+        else:
```
Comment (baskaryan) [inline libs/langchain/langchain/retrievers/you.py:49]: slight preference to make this an explicit check for a value (eg. `elif self.endpoint_type == "snippet"`) and to raise an error if value is unknown

**Review by baskaryan — COMMENTED** [review]
two small comments but overall looks great!

File: libs/langchain/langchain/retrievers/you.py (hunk 1 above)
Comment (sam-h-bean) [reply libs/langchain/langchain/retrievers/you.py]: changed

**Pull request merged**